lib/babe/build.go

```diff
@@ -270,6 +265,21 @@ func (b *BlockBuilder) buildBlockExtrinsics(slot Slot) []*transaction.ValidTrans
 			if _, ok := err.(*DispatchOutcomeError); !ok {
 				continue
 			}
+
+			// don't drop transactions that may be valid in a later block ie.
+			// run out of gas for this block or have a nonce that may be valid in a later block
+			var e *TransactionValidityError
+			if !errors.As(err, &e) {
+				continue
+			}
+
+			err = err.(*TransactionValidityError).msg
+			if errors.Is(err, errExhaustsResources) || errors.Is(err, errInvalidTransaction) {
```

Comment:
```suggestion
			if errors.Is(e, errExhaustsResources) || errors.Is(e, errInvalidTransaction) {
```

Reply:
not sure why this is necessary?

Reply:
it just changes to use the `e.msg` instead of `err` since `e` variable has the correct `msg` error

Reply:
updated